### PR TITLE
Feature/refactor shape

### PR DESCRIPTION
**Pull request by michelengelen**
#### Summary
refactored shape component and adjusted some styles, etc. on the way there

### CODE REVIEW

File: src/foundations/shape/Shape.tsx [hunk 1]
```diff
@@ -1,47 +1,63 @@
...
-        <StyledShape as={component} {...styledShapeProperties}>
-            {children}
-        </StyledShape>
-    );
+    return '';
 };
 
+const Shape = styled.div.withConfig({
+    // this is to prevent all properties to being passed down to the underlying
+    // component, except for the ones we want to having them passe down
```
Comment (nevyangelova) [inline src/foundations/shape/Shape.tsx:42]: nit: typos

File: src/foundations/theme-provider/global-styles/globalStyles.ts [hunk 2]
```diff
@@ -9,12 +9,15 @@ import VElevation from './variables.elevation';
...
 import VFontStyle from './variables.font-style';
 import VBorderRadius from './variables.border-radius';
+import reset from './reset-styles';
 
 type PGlobalStyles = {
     theme: TTheme;
 };
 
 const GlobalStyle = createGlobalStyle`
+    ${reset};
```
Comment (nevyangelova) [inline src/foundations/theme-provider/global-styles/globalStyles.ts:19]: When would we want to reset the global styles?

File: src/foundations/layout/Grid.props.ts [hunk 3]
```diff
@@ -25,6 +27,10 @@ type PGrid = {
     padding?: TSpacingDefinition;
     /** margin according to `TSpacingDefinition` type */
     margin?: TSpacingDefinition;
+    /** restrict the width of a `Grid` */
+    width?: number | null;
+    /** restrict the height of a `Grid` */
+    height?: number | null;
     children?: React.ReactNode;
```
Comment (nevyangelova) [inline src/foundations/layout/Grid.props.ts:34]: I am still 3/5 on positioning the Shape component relatively when we render child components inside it and not having to rely on rendering another Grid component inside.

File: src/foundations/theme-provider/global-styles/globalStyles.ts (hunk 2 above)
Comment (michelengelen) [reply src/foundations/theme-provider/global-styles/globalStyles.ts:19]: this is to reset all styles that come from the browsers.

File: src/foundations/layout/Grid.props.ts (hunk 3 above)
Comment (michelengelen) [reply src/foundations/layout/Grid.props.ts:34]: and you are right ... we currently do that
This is for another use-case. :D

File: src/foundations/shape/Shape.tsx (hunk 1 above)
Comment (michelengelen) [reply src/foundations/shape/Shape.tsx:42]: done

**Review by nevyangelova — APPROVED** [review]

**Pull request merged**